### ISSUE TITLE
Is there a way I can deploy my app without wiping my MongoDB on the production server?

### ISSUE THREAD

**Issue by sproleee** [arunoda/meteor-up-legacy#972]
I'm installing my SSL certificate and need to run `mupx setup` and `mupx deploy` to get it to work.

However, doing this and then deploying my app will wipe my MongoDB on my production server and users lose all of their accounts. It would be a nightmare.

I simply run `mupx deploy` after making fixed to my Meteor app to update my app. However, when installing my SSL certificate, I have to run `mupx setup` to get it to work.

Is there a way I can do this without wiping my user database on my deployment server?

**Comment by nilsi** [comment]
@sproleee after reading this I got worried what I shall do when I need to update my cert. Have you got any further with this?

**Comment by sproleee** [comment]
@nilsi The easiest way is to back up your MongoDB on production server, `mupx setup` and `mupx deploy` and then replace the new DB (empty one) with the backup.

My only problem now is `mongodump` doesn't properly back up my MongoDB on DigitalOcean, so I'm trying to figure out what's wrong.

**Comment by MasterJames** [comment]
I did a Mongo backup and restore PR for mupx back in Nov maybe it will help.

https://github.com/arunoda/meteor-up/pull/736

You should be able to overwrite the cert in the persistent folder that docker run loads via the volume flag with your new ones but I believe the long term the plan if not added already is to use letsencrypt automation in the new port at kadirahq/meteor-up.

**Comment by nilsi** [comment]
@sproleee Okey I have used 3T MongoChef to make backups it works very well I think. You might wanna look into that. Just connect to your DO instance. 

@MasterJames Okey thanks! So I should be able to SSH into the machine. Then override the cert and it should work without running mupx setup? Will have to do some research on how to do this.

**Comment by bensventures** [comment]
Are you sure running `mupx setup` will wipe the mongo database? I know that, at least with `mup`, you can run setup and it will keep the data in the db. I've done it multiple times successfully. I haven't tried with `mupx` so this is obviously at your own risks.

From the `mupx` example config file:

```
// Install MongoDB on the server. Does not destroy the local MongoDB on future setups
  "setupMongo": true,
```

**Comment by MasterJames** [comment]
Yes it's how to overwrite on the target from your devbox. Mupx uses docker and stomps over the deployment targets persistent local file store the docker container mounts via the volume flag when deploying.

**Comment by massimosgrelli** [comment]
@bensventures the same for me. Just done mup setup + mup deploy after I added nginx to my configuration and everything worked fine. My database is still there. Does mupx act differently because of containers?

**Comment by hyperh** [comment]
@nilsi Did you figure out how to override the cert in the docker container manually? This is what I found so far, just wondering if you managed to do it. http://stackoverflow.com/questions/26028971/docker-container-ssl-certificates

**Comment by nilsi** [comment]
@hyperh I spend the whole day yesterday messing around with MUPX. I did both `mup setup` and `mup deploy` multiple times after I took a backup but it never wiped my database on the server. I then migrated to [kadirahq:meteor-up](https://github.com/kadirahq/meteor-up)  which worked fine and never managed to accidentally wipe the DB. There was no problem updating the cert for me. 

However I recommend you to take a backup before you starting doing this to be sure.

**Comment by hyperh** [comment]
@nilsi do you mean mup setup or mupx setup didn't wipe your mongo db?

**Comment by nilsi** [comment]
Oh both actually. But on your on risk. Seems like @MasterJames had a different experience.

It does not touch the Mongo container if it is already there it seems like. However the other app containers getting wiped every time.

**Comment by hyperh** [comment]
@nilsi Do you have any resources on how to back up and restore a mongo db on a mupx container?

**Comment by nilsi** [comment]
Not more than what's in this thread already. I used  3T MongoChef to take a backup and saved to my local machine, worked out well.

**Comment by hyperh** [comment]
@nilsi Hmm, I can't seem to connect to my app's mongodb using 3T MongoChef. 
I tried this:

```
myapp.com:27017
```

Getting this error

```
Timed out after 30000 ms while waiting for a server that matches ReadPreferenceServerSelector{readPreference=primary}. Client view of cluster state is {type=UNKNOWN, servers=[{address=myapp.com:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by

 {java.net.ConnectException: Connection refused}}]
```

Probably because I've already deployed it to Digital Ocean using mupx. Not sure how you are connecting to it.

mupx docs say https://github.com/arunoda/meteor-up/tree/mupx

> If you decided to use MongoDB, it'll be also running as a Docker conatiner. It's bound to the local interface and port 27017 (you cannot access from the outside)

**Comment by nilsi** [comment]
When setting up the connection you should probably use localhost as server IP. And setup SSH under the SSH Tunnel tab where you use the real IP of the DO server.

**Server tab:**
IP : 127.0.0.1 PORT : 27017
**SSH Tunnel tab:**
SSH Address: DO ip address Port: 22
And fill in username and password

If it doesn't work SSH into the server and run `docker ps` to see what port your database running on.

**Comment by hyperh** [comment]
@nilsi It worked! I managed to connect. Thanks a lot for your help. I'll see if I can backup and restore using it now.

**Comment by nilsi** [comment]
Great! Good luck!
